stylelint-polaris/plugins/typography-declaration-property-value-disallowed-list/index.js

```diff
@@ -0,0 +1,51 @@
+const stylelint = require('stylelint');
+
+const ruleName =
+  'polaris/typography-declaration-property-value-disallowed-list';
+
+/**
+ * @typedef {{
+ *   [property: string]: string | RegExp | (string | RegExp)[]
+ * }} PrimaryOptions
+ */
+
+const {rule} = stylelint.createPlugin(
+  ruleName,
+  /** @param {PrimaryOptions} primary */
+  (primary) => {
+    return (root, result) => {
+      stylelint.utils.checkAgainstRule(
+        {
+          ruleName: 'declaration-property-value-disallowed-list',
+          ruleSettings: primary,
+          root,
+        },
+        (warning) => {
+          // Ignore invalid options as they are already surfaced by Stylelint
+          if (warning.stylelintType === 'invalidOption') return;
+
+          if (
+            warning.node.type === 'decl' &&
+            warning.node.prop === 'font-weight' &&
+            warning.node.parent.type === 'atrule' &&
+            warning.node.parent.name === 'font-face'
```

Comment:
Open question: Should this rule ignore all failures in `@font-face` rules?

There are [many descriptors](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face#descriptors) in `@font-face` rules and none of which can accept Polaris custom properties.

Reply:
I would say "yes". Perhaps just remove the `warning.node.prop === 'font-weight' &&` line so we can return early for all `@font-face` properties.